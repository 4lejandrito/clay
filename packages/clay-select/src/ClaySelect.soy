{namespace ClaySelect}

/**
 * This renders the component's whole content.
 */
{template .render}
	{@param options: list<?>}
	{@param? disabled: bool}
	{@param? id: string}
	{@param? inline: bool}
	{@param? label: string}
	{@param? multiple: bool}
	{@param? name: string}

	<div class="form-group" id="{$id}">
		{if $label}
			<label>{$label}</label>
		{/if}

		{let $select kind="html"}
			{call .select}
				{param disabled: $disabled /}
				{param multiple: $multiple /}
				{param name: $name /}
				{param options: $options /}
			{/call}
		{/let}

		{if $inline}
			<div class="form-inline">
				{$select}
			</div>
		{else}
			{$select}
		{/if}
	</div>
{/template}

/**
 * This renders the select tag.
 */
{template .select}
	{@param options: list<?>}
	{@param? disabled: bool}
	{@param? multiple: bool}
	{@param? name: string}

	{let $selectAttributes kind="attributes"}
		class="form-control"
		{if $disabled}
			disabled="disabled"
		{/if}
		{if $multiple}
			multiple="multiple"
		{/if}
		{if $name}
			name="{$name}"
		{/if}
	{/let}

	<select {$selectAttributes}>
		{foreach $option in $options}
			{let $optionAttributes kind="attributes"}
				{if $option.selected}
					selected="selected"
				{/if}
				value="{$option.value}"
			{/let}

			<option {$optionAttributes}>{$option.label}</option>
		{/foreach}
	</select>
{/template}
