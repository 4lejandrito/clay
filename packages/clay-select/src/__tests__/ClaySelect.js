'use strict';

import ClaySelect from '../ClaySelect';

let select;

describe('ClaySelect', function() {
	afterEach(() => {
		if (select) {
			select.dispose();
		}
	});

	it('should generate a select element', function() {
		select = new ClaySelect({
			options: [
				{
					label: 'Option 1',
					value: '1',
				},
				{
					label: 'Option 2',
					value: '2',
				},
			],
		});

		expect(select).toMatchSnapshot();
	});

	it('should generate a select element with label', function() {
		select = new ClaySelect({
			label: 'My Select',
			options: [
				{
					label: 'Option 1',
					value: '1',
				},
				{
					label: 'Option 2',
					value: '2',
				},
			],
		});

		expect(select).toMatchSnapshot();
	});

	it('should generate a select element with name', function() {
		select = new ClaySelect({
			name: 'myselectname',
			options: [
				{
					label: 'Option 1',
					value: '1',
				},
				{
					label: 'Option 2',
					value: '2',
				},
			],
		});

		expect(select).toMatchSnapshot();
	});

	it('should generate a select element with id', function() {
		select = new ClaySelect({
			id: 'myselectid',
			options: [
				{
					label: 'Option 1',
					value: '1',
				},
				{
					label: 'Option 2',
					value: '2',
				},
			],
		});

		expect(select).toMatchSnapshot();
	});

	it('should generate an inline select element', function() {
		select = new ClaySelect({
			inline: true,
			options: [
				{
					label: 'Option 1',
					value: '1',
				},
				{
					label: 'Option 2',
					value: '2',
				},
			],
		});

		expect(select).toMatchSnapshot();
	});

	it('should generate an inline select element with label', function() {
		select = new ClaySelect({
			inline: true,
			label: 'My Select',
			options: [
				{
					label: 'Option 1',
					value: '1',
				},
				{
					label: 'Option 2',
					value: '2',
				},
			],
		});

		expect(select).toMatchSnapshot();
	});

	it('should generate a select element with preselected option', function() {
		select = new ClaySelect({
			options: [
				{
					label: 'Option 1',
					value: '1',
				},
				{
					label: 'Option 2',
					selected: true,
					value: '2',
				},
			],
		});

		expect(select).toMatchSnapshot();
	});

	it('should generate a disabled select element', function() {
		select = new ClaySelect({
			disabled: true,
			options: [
				{
					label: 'Option 1',
					value: '1',
				},
				{
					label: 'Option 2',
					value: '2',
				},
			],
		});

		expect(select).toMatchSnapshot();
	});

	it('should generate a multiple select element', function() {
		select = new ClaySelect({
			multiple: true,
			options: [
				{
					label: 'Option 1',
					value: '1',
				},
				{
					label: 'Option 2',
					value: '2',
				},
			],
		});

		expect(select).toMatchSnapshot();
	});

	it('should generate an inline multiple select element', function() {
		select = new ClaySelect({
			inline: true,
			multiple: true,
			options: [
				{
					label: 'Option 1',
					value: '1',
				},
				{
					label: 'Option 2',
					value: '2',
				},
			],
		});

		expect(select).toMatchSnapshot();
	});
});
