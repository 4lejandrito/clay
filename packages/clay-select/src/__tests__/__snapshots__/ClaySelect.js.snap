// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClaySelect should generate a disabled select element 1`] = `
<div class="form-group">
  <select class="form-control" disabled="disabled">
    <option value="1">Option 1</option>
    <option value="2">Option 2</option>
  </select>
</div>
`;

exports[`ClaySelect should generate a multiple select element 1`] = `
<div class="form-group">
  <select class="form-control" multiple="multiple">
    <option value="1">Option 1</option>
    <option value="2">Option 2</option>
  </select>
</div>
`;

exports[`ClaySelect should generate a select element 1`] = `
<div class="form-group">
  <select class="form-control">
    <option value="1">Option 1</option>
    <option value="2">Option 2</option>
  </select>
</div>
`;

exports[`ClaySelect should generate a select element with id 1`] = `
<div class="form-group" id="myselectid">
  <select class="form-control">
    <option value="1">Option 1</option>
    <option value="2">Option 2</option>
  </select>
</div>
`;

exports[`ClaySelect should generate a select element with label 1`] = `
<div class="form-group">
  <label>My Select</label>
  <select class="form-control">
    <option value="1">Option 1</option>
    <option value="2">Option 2</option>
  </select>
</div>
`;

exports[`ClaySelect should generate a select element with name 1`] = `
<div class="form-group">
  <select class="form-control" name="myselectname">
    <option value="1">Option 1</option>
    <option value="2">Option 2</option>
  </select>
</div>
`;

exports[`ClaySelect should generate a select element with preselected option 1`] = `
<div class="form-group">
  <select class="form-control">
    <option value="1">Option 1</option>
    <option selected="selected" value="2">Option 2</option>
  </select>
</div>
`;

exports[`ClaySelect should generate an inline multiple select element 1`] = `
<div class="form-group">
  <div class="form-inline">
    <select class="form-control" multiple="multiple">
      <option value="1">Option 1</option>
      <option value="2">Option 2</option>
    </select>
  </div>
</div>
`;

exports[`ClaySelect should generate an inline select element 1`] = `
<div class="form-group">
  <div class="form-inline">
    <select class="form-control">
      <option value="1">Option 1</option>
      <option value="2">Option 2</option>
    </select>
  </div>
</div>
`;

exports[`ClaySelect should generate an inline select element with label 1`] = `
<div class="form-group">
  <label>My Select</label>
  <div class="form-inline">
    <select class="form-control">
      <option value="1">Option 1</option>
      <option value="2">Option 2</option>
    </select>
  </div>
</div>
`;
